tests/integration/ceph_bucket_notification_test.go

```diff
@@ -122,38 +110,165 @@ func (s *ObjectSuite) TestBucketNotificationsInOrder() {
+		})
+		assert.True(t, notificationPresent)
+	})
 
+	t.Run("removing label from the ObjectBucketClaim", func(t *testing.T) {
+		obc, err := k8sh.BucketClientset.ObjectbucketV1alpha1().ObjectBucketClaims(obcNamespace).Get(ctx, obcName, metav1.GetOptions{})
 		assert.Nil(t, err)
-		logger.Infof("endpoint (%s) Accesskey (%s) secret (%s) region (%s)", s3endpoint, s3AccessKey, s3SecretKey, rgwcontext.ZoneGroup)
+		delete(obc.Labels, "test-label")
+		_, err = k8sh.BucketClientset.ObjectbucketV1alpha1().ObjectBucketClaims(obcNamespace).Update(ctx, obc, metav1.UpdateOptions{})
+		assert.Nil(t, err)
+		// check whether existing bucket notification uneffected
+		notificationPresent := utils.Retry(12, 2*time.Second, "notification is created for bucket", func() bool {
+			return helper.BucketClient.CheckBucketNotifications(namespace, storeName, obcName, bucketname, notificationName, s3endpoint)
+		})
+		assert.True(t, notificationPresent)
+	})
 
-		created := utils.Retry(12, 2*time.Second, "notification is created for bucket", func() bool {
-			notifications, err := s3client.Client.GetBucketNotificationConfiguration(&s3.GetBucketNotificationConfigurationRequest{
-				Bucket: &bucketname,
-			})
-			if err != nil {
-				logger.Infof("failed to fetch bucket notifications configuration due to %v", err)
-				return false
+	t.Run("remove notification from the ObjectBucketClaim", func(t *testing.T) {
+		obc, err := k8sh.BucketClientset.ObjectbucketV1alpha1().ObjectBucketClaims(obcNamespace).Get(ctx, obcName, metav1.GetOptions{})
+		assert.Nil(t, err)
+		delete(obc.Labels, bucketNotificationLabelPrefix+notificationName)
+		_, err = k8sh.BucketClientset.ObjectbucketV1alpha1().ObjectBucketClaims(obcNamespace).Update(ctx, obc, metav1.UpdateOptions{})
+		assert.Nil(t, err)
+		// check whether existing bucket notification uneffected
+		var notificationPresent bool
+		for i := 0; i < 4; i++ {
+			notificationPresent = helper.BucketClient.CheckBucketNotifications(namespace, storeName, obcName, bucketname, notificationName, s3endpoint)
+			if !notificationPresent {
+				break
 			}
-			logger.Infof("%d bucket notifications found in: %v", len(notifications.TopicConfigurations), notifications)
-			for _, notification := range notifications.TopicConfigurations {
-				if *notification.Id == notificationName {
-					return true
-				}
-				logger.Infof("bucket notifications name mismatch %q != %q", *notification.Id, notificationName)
+			time.Sleep(5 * time.Second)
+		}
+		assert.False(t, notificationPresent)
+	})
+
+	t.Run("add new topic, notification to existing ObjectBucketClaim", func(t *testing.T) {
+		newNotificationName := "new-notification"
+		newTopicName := "new-topic"
+		err := helper.TopicClient.CreateTopic(newTopicName, storeName, httpEndpointService)
+		assert.Nil(t, err)
+		created := utils.Retry(12, 2*time.Second, "topic is created", func() bool {
+			return helper.TopicClient.CheckTopic(newTopicName)
+		})
+		assert.True(t, created)
```

Comment:
Let's use more subtests like we do in the object test. That way, the steps show up in the output, and the sections are clearer to read at a glance. This section could be enclosed in `t.Run("create and verify topic", ...)`, for example.